### PR TITLE
Support set/hashset/treeSet/arraylist/linkedlist types in the agent and plugin configurations

### PR DESCRIPTION
**Pull request by pg-yang**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

Set is faster than List for detecting value if exist . and could remove  duplicate  value  too

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Release Notes.
 8.13.0
 ------------------
 
+* Config support Set parameter
```
Comment (wu-sheng) [inline CHANGES.md:8]:
```suggestion
* support set-type in the agent or plugin configurations.
```

**Comment by wu-sheng** [comment]
Set is a generic type, I noticed you only support String Set? 
You should read the generic type and convert element type too.

**Comment by wu-sheng** [comment]
Set is a generic type, I noticed you only support String Set? 
You should read the generic type and convert element type too.

**Comment by pg-yang** [comment]
Try to  support Set<String> , String<Integer>  , etc .

**Comment by wu-sheng** [comment]
I mean the codes are converting to set string forcely, no matter what generic type declared.

**Comment by pg-yang** [comment]
Are  you  mean  Java-generics ? like  Set<String> , Set<Integer> ....

**Comment by wu-sheng** [comment]
Yes, based on your codes, I believe you still would add string in to int set.
We should convert configuration to the expected type.

**Comment by pg-yang** [comment]
OK, I'll support all `java.util.Collection` . It's better

**Comment by pg-yang** [comment]
PING

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**